backend/src/main/java/gov/cdc/usds/simplereport/config/GraphQlConfig.java

```diff
@@ -0,0 +1,86 @@
+package gov.cdc.usds.simplereport.config;
+
+import static java.util.Collections.singletonList;
+
+import gov.cdc.usds.simplereport.api.DefaultArgumentValidation;
+import gov.cdc.usds.simplereport.api.directives.RequiredPermissionsDirectiveWiring;
+import gov.cdc.usds.simplereport.api.model.errors.ConflictingUserException;
+import gov.cdc.usds.simplereport.api.model.errors.GenericGraphqlException;
+import gov.cdc.usds.simplereport.api.model.errors.IllegalGraphqlArgumentException;
+import gov.cdc.usds.simplereport.api.model.errors.IllegalGraphqlFieldAccessException;
+import gov.cdc.usds.simplereport.api.model.errors.NonexistentUserException;
+import gov.cdc.usds.simplereport.config.scalars.UploadScalarType;
+import gov.cdc.usds.simplereport.config.scalars.datetime.DateTimeScalar;
+import gov.cdc.usds.simplereport.config.scalars.localdate.LocalDateScalar;
+import graphql.GraphQLError;
+import graphql.validation.rules.OnValidationErrorStrategy;
+import graphql.validation.rules.ValidationRules;
+import graphql.validation.schemawiring.ValidationSchemaWiring;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.graphql.execution.DataFetcherExceptionResolver;
+import org.springframework.graphql.execution.RuntimeWiringConfigurer;
+import org.springframework.security.access.AccessDeniedException;
+import reactor.core.publisher.Mono;
+
+@Slf4j
+@Configuration
+public class GraphQlConfig {
+  public static final String REQUIRED_PERMISSIONS_DIRECTIVE_NAME = "requiredPermissions";
+  public static final int MAXIMUM_SIZE = 256;
+
+  // mask all exception by default
+  // to customize errors returned, you have to manually throw it here
+  @Bean
+  public DataFetcherExceptionResolver dataFetcherExceptionResolver() {
```

Comment:
replaces [this good](https://github.com/CDCgov/prime-simplereport/pull/4243/files#diff-ec7f4d4bc1ab8722993d0d653267c31980ee2413059326fc6166dedab5ceaa32)

Reply:
non-blocking question: what's the motivation behind wrapping all exceptions as GenericGraphQLExceptions? if we don't do this, do they not get included in the errors section of the response?

Reply:
Mainly security, we don't want to send a trace back to the client since that is not necessary